PowerShellTools.HostService/ServiceManagement/Debugging/PowerShellDebuggingService.cs

```diff
@@ -140,6 +140,34 @@ public void SetRunspace(bool overrideExecutionPolicy)
         }
 
         /// <summary>
+        /// Examines a process' modules for powershell.exe which indicates the process should
+        /// be attachable. Will generate an exception if the host process is x86 and the process
+        /// is x64.
+        /// </summary>
+        /// <param name="pid"></param>
+        /// <returns>True if powershell.exe is a module of the process, false otherwise.</returns>
+        public bool IsAttachable(uint pid)
+        {
+            try
+            {
+                var process = Process.GetProcessById((int)pid);
+                ServiceCommon.Log("IsAttachable:" + process.ProcessName + "; id:" + process.Id);
+                foreach (ProcessModule module in process.Modules)
+                {
+                    if (module.ModuleName.Equals("powershell.exe", StringComparison.Ordinal) && _installedPowerShellVersion >= RequiredPowerShellVersionForProcessAttach)
+                    {
+                        return true;
+                    }
+                }
+            }
+            catch (Exception ex)

```

Comment:
This method was actually present in the local attach code. I simply moved its location to be more consistent with how I was ordering these functions. But to answer your question, yes we do. See the summary comment: "Will generate an exception if the host process is x86 and the process is x64."